torch/nn/parallel/distributed.py

```diff
@@ -117,14 +122,34 @@ def __init__(self, module, device_ids=None, output_device=None, dim=0):
         else:
             self._module_copies = [self.module]
 
+        # Split parameters into type buckets so that parameter sync (broadcast)
+        # can operates on mixed parameter types. (e.g. mixed half and float)
+        self.param_type_buckets = {}
+        for device_idx, module in enumerate(self._module_copies):
+            for p in module.parameters():
+                tp = type(p.data)
```

Comment:
you need to use `p.type()`. Otherwise it won't work once Variables and Tensors are merged

Reply:
Will do